api/tests/integration/infrastructure/repositories/organization-invited-user-repository_test.js

```diff
@@ -0,0 +1,473 @@
+
+describe('Integration | Repository | OrganizationInvitedUserRepository', function () {
+  afterEach(async function () {
+    await knex('user-orga-settings').delete();
+    await knex('memberships').delete();
+  });
+
+  describe('#get', function () {
+    it('should return an OrganizationInvitedUser userId', async function () {
+      // given
+      const ExpectedUserId = 123;
+      const organizationInvitationId = databaseBuilder.factory.buildOrganizationInvitation().id;
+      databaseBuilder.factory.buildUser({
+        email: 'wrongUser@example.net',
+      });
+      const user = databaseBuilder.factory.buildUser({
+        id: ExpectedUserId,
+        email: 'user@example.net',
+      });
+
+      await databaseBuilder.commit();
+
+      // when
+      const OrganizationInvitedUser = await organizationInvitedUserRepository.get({
+        organizationInvitationId,
+        email: user.email,
+      });
+
+      // then
+      expect(OrganizationInvitedUser.userId).to.equal(ExpectedUserId);
+    });
+
+    it('should return an OrganizationInvitedUser.invitation', async function () {
+      // given
+      const organizationInvitation = databaseBuilder.factory.buildOrganizationInvitation();
+      const user = databaseBuilder.factory.buildUser({
+        email: 'user@example.net',
+      });
+      await databaseBuilder.commit();
+
+      const expectedOrganizationInvitation = {
+        id: organizationInvitation.id,
+        organizationId: organizationInvitation.organizationId,
+        code: organizationInvitation.code,
+        role: organizationInvitation.role,
+      };
+
+      // when
+      const OrganizationInvitedUser = await organizationInvitedUserRepository.get({
+        organizationInvitationId: organizationInvitation.id,
+        email: user.email,
+      });
+
+      // then
+      expect(OrganizationInvitedUser.invitation).to.deep.equal(expectedOrganizationInvitation);
+    });
+
+    describe('when memberships exist', function () {
+      it('should return an OrganizationInvitedUser.currentMembershipId', async function () {
+        // given
+        const organizationInvitation = databaseBuilder.factory.buildOrganizationInvitation();
+        const user = databaseBuilder.factory.buildUser({
+          email: 'user@example.net',
+        });
+        const memberships = databaseBuilder.factory.buildMembership({
+          userId: user.id,
+          organizationId: organizationInvitation.organizationId,
+        });
+        await databaseBuilder.commit();
+
+        // when
+        const OrganizationInvitedUser = await organizationInvitedUserRepository.get({
+          organizationInvitationId: organizationInvitation.id,
+          email: user.email,
+        });
+
+        // then
+        expect(OrganizationInvitedUser.currentMembershipId).to.equal(memberships.id);
+      });
+
+      it('should return an OrganizationInvitedUser.currentRole', async function () {
```

Comment:
idem pour les `it` suivant ... `should return the current role of the invited user`